src/libslic3r/GCodeWriter.cpp

```diff
@@ -222,9 +224,15 @@ std::string GCodeWriter::set_acceleration_internal(Acceleration type, unsigned i
 
 std::string GCodeWriter::set_jerk_xy(double jerk)
 {
+    double jerk_x = jerk;
+    double jerk_y = jerk;
+    
     // Clamp the jerk to the allowed maximum.
-    if (m_max_jerk > 0 && jerk > m_max_jerk)
-        jerk = m_max_jerk;
+    if (m_max_jerk_x > 0 && jerk > m_max_jerk_x)
```

Comment:
We still need to cap jerk, it's used by Klipper in below.

Reply:
Like this?
```
    m_last_jerk = jerk;
    jerk = std::min(jerk, std::min(jerk_x, jerk_y));
```

Reply:
It's better to move the logic inside following block respectively to make it cleaner as Klipper don't use jerk_x/y.
```
if(FLAVOR_IS(gcfKlipper))
        gcode << "SET_VELOCITY_LIMIT SQUARE_CORNER_VELOCITY=" << jerk;
    else
        gcode << "M205 X" << jerk_x << " Y" << jerk_y;
```